### PR TITLE
Fixes #25730 - only assign many to many associations

### PR DESCRIPTION
**Pull request by jlsherrill**
for taxonomies during db seed

### CODE REVIEW

File: db/seeds.d/050-taxonomies.rb [hunk 1]
```diff
@@ -8,7 +8,11 @@
     taxonomy.without_auditing do
       tax_name = ENV.fetch("SEED_#{taxonomy.to_s.upcase}", "Default #{taxonomy}")
       tax = taxonomy.create!(name: tax_name)
-      associations = taxonomy.reflect_on_all_associations.reject {|assoc| skip_associations.include?(assoc.name)}
+      associations = associations.select do |assoc|
+        !skip_associations.include?(assoc.name) &&
+        (assoc.is_a?(ActiveRecord::Reflection::HasManyReflection) ||
+            assoc.is_a?(ActiveRecord::Reflection::HasAndBelongsToManyReflection))
+      end
```
Comment (jlsherrill) [inline db/seeds.d/050-taxonomies.rb:15]: I thought about just making sure this isn't a HasOne, but that wouldn't catch a BelongsTo (which probably doesn't exist, but may).  Also, this seems to be assigning both the HasMany and HasMany Through relationships, which seemed needless.

**Comment by beav** [comment]
@jlsherrill is the katello test failure expected, since this has to be merged before the katello fix?

File: db/seeds.d/050-taxonomies.rb (hunk 1 above)
Comment (tbrisker) [reply db/seeds.d/050-taxonomies.rb:15]: most associations are using `:through => :taxable_taxonomies`, so if we skip the `ActiveRecord::Reflection::ThroughReflection` ones we won't assign any of them. I'd rather we skip `HasOne` and `BelongsTo` specifically.

**Review by tbrisker — CHANGES_REQUESTED** [review]

**Comment by jlsherrill** [comment]
@tbrisker works for me! I figured given two options i'd pick the one that wasn't desired ;)

Updated!

@beav, i believe so, yes

File: db/seeds.d/050-taxonomies.rb [hunk 2]
```diff
@@ -8,7 +8,11 @@
     taxonomy.without_auditing do
       tax_name = ENV.fetch("SEED_#{taxonomy.to_s.upcase}", "Default #{taxonomy}")
       tax = taxonomy.create!(name: tax_name)
-      associations = taxonomy.reflect_on_all_associations.reject {|assoc| skip_associations.include?(assoc.name)}
+      associations = taxonomy.reflect_on_all_associations.select do |assoc|
```
Comment (tbrisker) [inline db/seeds.d/050-taxonomies.rb]: How about applying a De Morgan here and making it a reject instead? Not really crucial, but might make it a bit clearer to understand.

Comment (jlsherrill) [reply db/seeds.d/050-taxonomies.rb]: works for me!

**Review by tbrisker — APPROVED** [review]

**Comment by tbrisker** [comment]
Thanks @jlsherrill !

**Pull request merged**